components/screens/Registration.tsx

```diff
@@ -27,45 +26,71 @@ export default function Register({ navigation }: Props) {
     // first, check that the username is unique
     // if it's unique, then try to create the new auth account
     setError({})
+    const { error } = await supabase.auth.signUp({
+      email: email,
+      password: password,
+    })
 
-    await firestore()
-      .collection('users')
-      .where('name', '==', username)
-      .get()
-      .then(querySnapshot => {
-        // querySnapshot.empty still true if size is 0...?
-        if (querySnapshot.size > 0) {
-          return setError({userName: 'That username is taken!'})
-        }
-        auth()
-          .createUserWithEmailAndPassword(email, password)
-          .then((user: FirebaseAuthTypes.UserCredential) => {
-            // this will automatically send the user to /Lobby
-            // now try to connec the auth account to a User record
-            firestore()
-              .collection('users')
-              .add({
-                name: username,
-                email: email,
-                user: user.user.uid,
-              } as FirestoreUser)
-              .then(() => {
-                console.log('User added!')
-              }).catch((err) => {
-                console.log(err)
-                // todo
-              })
-            console.log('User account created & signed in!')
-          })
-          .catch(error => {
-            if (error.code === 'auth/email-already-in-use') {
-              setError({email: 'That email address is already in use!'})
-            }
-            if (error.code === 'auth/invalid-email') {
-              setError({email: 'That email address is invalid!'})
-            }
-          })
-      })
+    if (error) {
+      //function to make simple alert
+      Alert.alert(
+        'Login failed', // alert title
+        error.message, // alert body
+        [
+          { text: 'OK', onPress: () => console.log('OK Pressed') }
+        ],
+        { cancelable: true }
+      )
+    } else {
+      //function to make simple alert
+      Alert.alert(
+        'Success', // alert title
+        'Please check your email to verify your account before logging in', // alert body
+        [
+          { text: 'OK', onPress: () => navigation.navigate('Login')}
+        ],
+        { cancelable: true }
+      )
+    }
+
+    // await firestore()
```

Comment:
Do we still need the old firestore code for reference?

Reply:
I'm fine with leaving it in for now so we can reference the logic later. #notsureif it's gonna help, but it doesn't hurt to leave it for now